toolbox/src/InverseDynamics.cpp

```diff
@@ -139,20 +134,15 @@ bool InverseDynamics::initialize(BlockInformation* blockInfo)
     using namespace iDynTree;
 
     // Get the DoFs
-    const auto robotInterface = getRobotInterface(blockInfo).lock();
-    if (!robotInterface) {
-        wbtError << "RobotInterface has not been correctly initialized.";
-        return false;
-    }
-    const auto dofs = robotInterface->getConfiguration().getNumberOfDoFs();
+    const auto dofs = getRobotInterface()->getConfiguration().getNumberOfDoFs();
```

Comment:
Nitpick, not blocking the merging of the PR: I think that a good old `const size_t dofs` would be much more readable here and in other places.